### PR TITLE
[Storage-preview] Bug fixes for ADLS Gen2

### PR DESCRIPTION
**Pull request by Juliehzl**
---
- `az storage blob list` list with directory flag
- `az storage blob move` add blob check for source
- `az storage blob directory move` add directory check for source and add --move-mode
- `az storage blob directory create` refine arguments name and help message

This checklist is used to make sure that common guidelines for a pull request are followed.

### General Guidelines

- [ ] Have you run `azdev style <YOUR_EXT>` locally? (`pip install azdev` required)
- [ ] Have you run `python scripts/ci/test_index.py -q` locally?

For new extensions:

- [ ] My extension description/summary conforms to the [Extension Summary Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/extensions/extension_summary_guidelines.md).

### CODE REVIEW

**Comment by jsntcy** [comment]
Which bugs does this PR fix? Could you please add more details in PR description?

File: src/storage-preview/azext_storage_preview/_help.py [hunk 1]
```diff
@@ -207,6 +207,8 @@
     examples:
         - name: Create a storage blob directory in a storage container.
           text: az storage blob directory create -c MyContainer -d MyDirectoryPath --account-name MyStorageAccount
+        - name: Create a storage blob directory with permissions and umask.
+          text: az storage blob directory create -c MyContainer -d MyDirectoryPath --account-name MyStorageAccount --permissions rwxrwxrwx --umask 0000
```
Comment (haroldrandom) [inline src/storage-preview/azext_storage_preview/_help.py]: Is it dup? --permision and --umask. 

I think they do the same job but just differentiate in format.

File: src/storage-preview/azext_storage_preview/_validators.py [hunk 2]
```diff
@@ -173,11 +174,25 @@ def validate_blob_directory_download_source_url(cmd, namespace):
...
+    destination = destination[1:] if destination.startswith('/') else destination
+    from azure.common import AzureException
+    try:
+        props = client.get_blob_properties(container, destination)
+        if not is_directory(props):
+            raise ValueError('usage error: You are specifying --destination-path with a blob name, not directory name. '
+                             'Please change to a valid blob directory name. If you want to upload to a blob file, '
+                             'please use `az storage blob upload` command.')
+    except AzureException:
+        pass
```
Comment (haroldrandom) [inline src/storage-preview/azext_storage_preview/_validators.py]: 1. Can AzureException catch ValueError?
2. Why let the exception pass?

File: src/storage-preview/azext_storage_preview/_help.py (hunk 1 above)
Comment (Juliehzl) [reply src/storage-preview/azext_storage_preview/_help.py]: No. They will cooperate. Th final permission setting will be permissions & ^umask.

File: src/storage-preview/azext_storage_preview/_validators.py (hunk 2 above)
Comment (Juliehzl) [reply src/storage-preview/azext_storage_preview/_validators.py]: 1. NO
2. When source doesn't exist, AzureException will be thrown but we can continue to create a new one for user. Therefore I let it pass.

File: src/storage-preview/azext_storage_preview/_help.py (hunk 1 above)
Comment (haroldrandom) [reply src/storage-preview/azext_storage_preview/_help.py]: emmm... Interesting...

**Pull request merged**